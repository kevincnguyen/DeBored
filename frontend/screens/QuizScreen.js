import React, { useState } from "react";
import { View, StyleSheet, TextInput, Text, Button } from "react-native";

import StartQuiz from "../components/Quiz/StartQuiz";
import QuizQuestion from "../components/Quiz/QuizQuestion";
import QuizResults from "../components/Quiz/QuizResults";

const quizQuestions = [
  {
    question: "Outdoor or Indoor?",
    options: ["Outdoor", "Indoor"],
  },
  {
    question: "Solo or Social?",
    options: ["Solo", "Social"],
  },
  {
    question: "Budget",
    options: ["$", "$$$"],
  },
];

const locationEntry = [  {
  question: "Enter your location or enter N/A if you would like not to"
}]

/*
 * The DeBored quiz screen that helps users find new activities.
 */
const QuizScreen = () => {
  const [quizStarted, setQuizStarted] = useState(false);
  const [currentQuestionIndex, setCurrentQuestionIndex] = useState(0);
  const [chosenAnswers, setChosenAnswers] = useState([]);
  const [locationInput, setLocationInput] = useState('');
  const [locationEntered, setLocationEntered] = useState(false)

  // Event handler for input change
  const handleLocationInputChange = (text) => {
    setLocationInput(text);
  };



  const handleStartPress = () => {
    setQuizStarted(true);
  };

  const processLocationInput = () => {
    setLocationEntered(true);
  };

  const handleResetPress = () => {
    setQuizStarted(false);
    setCurrentQuestionIndex(0);
    setChosenAnswers([]);
    setLocationEntered(false)
    setLocationInput('');
  };

  const handleNextQuestion = (option) => {
    setChosenAnswers([...chosenAnswers, option]);
    if (currentQuestionIndex < quizQuestions.length) {
      setCurrentQuestionIndex(currentQuestionIndex + 1);
    }
  };

  const renderQuizContent = () => {
    if (!quizStarted) {
      return <StartQuiz handleStartPress={handleStartPress} />;
    }

    if (currentQuestionIndex < quizQuestions.length) {
      return (
        <QuizQuestion
          question={quizQuestions[currentQuestionIndex]}
          handleNextQuestion={handleNextQuestion}
        />
      );
    } else if (!locationEntered) { 
  
      // render screen to get the location
      return (
      <View>
            <Text>Enter your location or enter N/A if you would like not to:</Text>
            <TextInput
              style={styles.input}
              placeholder="location"
              value={locationInput}
              onChangeText={handleLocationInputChange}
            />
            <Button title="Submit" onPress={processLocationInput} />
          </View>
      );
    } else {
      return (
        <QuizResults
          chosenAnswers={chosenAnswers}
          locationInput = {locationInput}
          handleResetPress={handleResetPress}
        />
      );
    }
  };

  return <View style={styles.container}>{renderQuizContent()}</View>;
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center", // Center the content both horizontally and vertically
    backgroundColor: "#FFFFFF",
  },
});

export default QuizScreen;
