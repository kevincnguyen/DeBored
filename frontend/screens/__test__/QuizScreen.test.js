import React from "react";
import renderer from "react-test-renderer";
import QuizScreen from "../QuizScreen";

// Mock child components
jest.mock("../../components/Quiz/StartQuiz", () => "StartQuiz");
jest.mock("../../components/Quiz/QuizQuestion", () => "QuizQuestion");
jest.mock("../../components/Quiz/QuizResults", () => "QuizResults");

jest.mock("react-native-paper", () => ({
  Provider: "Provider",
}));

jest.mock("react-native-paper-dropdown", () => ({
  DropDown: "DropDown",
}));

test("renders StartQuiz component initially", () => {
  const tree = renderer.create(<QuizScreen />).toJSON();
  expect(tree).toMatchSnapshot();
  expect(tree.children[0].children[0].type).toBe("StartQuiz");
});
