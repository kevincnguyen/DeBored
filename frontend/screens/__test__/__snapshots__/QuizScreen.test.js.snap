// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders StartQuiz component initially 1`] = `
<Provider>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <StartQuiz
      handleStartPress={[Function]}
    />
  </View>
</Provider>
`;
